MyBlog.Services/Areas/Users/UserService.cs

```diff
@@ -68,4 +86,43 @@ public async Task DeleteByIdAsync(int id)
         _context.Users.Remove(user);
         await _context.SaveChangesAsync();
     }
+    
+    private async Task<bool> CheckForFreeAsync(int id, UserDtoInput userInput)
+    {
+        var user = _context.Users.FirstOrDefault(u => u.Id == id)!;
+
```

Comment:
remove !

Reply:
also, you use async modificator in the method, why you don`t use FirstOrDefaultAsync overload, and don`t call await statement ?